### DISCUSSION TITLE
Svelte component testing - different directory

### DISCUSSION THREAD

**Discussion by magick93** [cypress-io/cypress#24651] [Questions and Help]
Hello

We are currently using cypress for e2e testing, and its working well.

We are interested in using cypress for testing our svelte components. However, we have our cypress project as a seperate project to our main svelte project.

Is it possible to configure the component testing to locate the components (and its project) in a project other than the cypress project?

**Comment by portyu9** [comment]
Hi @magick93 

Yes, but Cypress still needs the component source, its dependencies and its bundler configuration to be resolvable from the Cypress project.

For a separate Cypress workspace/repository, the cleanest setup is usually either:

- make both projects packages in the same workspace/monorepo, or
- explicitly provide the Svelte application's Vite/Webpack configuration to Cypress Component Testing.

Current Cypress supports explicit `viteConfig` and `webpackConfig` options, so you don't have to rely on automatic config discovery from the Cypress project's directory.

For example, conceptually:

```js
import appViteConfig from '../my-svelte-app/vite.config.js'

export default defineConfig({
  component: {
    devServer: {
      framework: 'svelte',
      bundler: 'vite',
      viteConfig: appViteConfig,
    },
  },
})
```

The important limitation is that imports used by the component must still resolve correctly — Svelte dependencies, aliases, CSS preprocessors, plugins, etc. all need to be available to the dev server.

`--project` only changes the Cypress project directory; it isn't a separate "component source root" setting.

Please mark this answer as accepted if it helped, thank you.
